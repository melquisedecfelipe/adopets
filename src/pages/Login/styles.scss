.login-container {
  display: flex;
  align-items: center;
  padding: 100px;
  height: 100vh;
  background-image: url('../../image/background.jpg');
  background-position: right bottom;
  background-size: cover;

  > form {
    display: flex;
    flex-direction: column;
    width: 500px;
    height: 450px;
    background-color: #ffffff;
    box-shadow: 0px 10px 30px #0000001a;
    border-radius: 5px;
    padding: 50px;

    > img {
      width: 170px;
    }

    > h2 {
      font-size: 32px;
      font-weight: bold;
      color: #160000;
      margin-top: 20px;
      margin-bottom: 0;
    }

    > p {
      font-size: 18px;
      font-weight: lighter;
      color: #160000;
      opacity: 0.6;
    }

    > input {
      border: 1px solid #8e879e;
      border-radius: 5px;
      height: 50px;
      margin-top: 10px;

      &:hover {
        border-color: #8e879e;
      }

      &:focus {
        border-right-width: 2px !important;
        border: 2px solid #8e879e;
        box-shadow: none;
      }
    }

    > button {
      width: 150px;
      height: 50px;
      border-radius: 5px;
      text-align: center;
      font-size: 18px;
      color: #ffffff;
      margin-top: 20px;
      padding: 10px;
    }

    .-rose {
      background-color: #ce3f71;

      &:hover {
        border-color: #ce3f71;
        color: #ffffff;
      }

      &:focus {
        border-color: #ce3f71;
      }
    }
  }
}

@media screen and (max-width: 1100px) {
  .login-container {
    justify-content: center;
  }
}
