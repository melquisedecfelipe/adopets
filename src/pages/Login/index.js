import React, { useState } from 'react';

import './styles.scss';

import Button from '../../components/Button';
import Input from '../../components/Input';

import api from '../../services/api';

import logo from '../../image/logo.svg';

export default function Login({ history }) {
  const [email, setEmail] = useState('');
  const [password, setPassword] = useState('');
  const sessionRequest = localStorage.getItem('session-request');

  async function handleSubmit(e) {
    e.preventDefault();
    try {
      const { data } = await api.post(
        'auth/session-register',
        {
          organization_user: {
            email,
            password,
          },
        },
        {
          headers: { Authorization: sessionRequest },
        },
      );

      localStorage.setItem('session-register', data.data.access_key);
      localStorage.setItem('organization-user', JSON.stringify(data.data.organization_user));
      history.push('/buscar');
    } catch (error) {
      console.error(error);
    }
  }

  function handleEmail(e) {
    setEmail(e.target.value);
  }

  function handlePassword(e) {
    setPassword(e.target.value);
  }

  return (
    <div className="login-container">
      <form>
        <img src={logo} alt="Adopets" />
        <h2>Bem-vindo(a)</h2>
        <p>Digite seu e-mail e senha para acessar o sistema</p>
        <Input type="text" placeholder="Seu e-mail" defaultValue={email} onBlur={handleEmail} />
        <Input
          type="password"
          placeholder="Sua senha"
          defaultValue={password}
          onBlur={handlePassword}
        />
        <Button classe="-rose" label="Entrar" onClick={handleSubmit} />
      </form>
    </div>
  );
}
