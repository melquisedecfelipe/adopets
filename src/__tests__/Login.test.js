import React from 'react';

import { shallow } from 'enzyme';

import Login from '../pages/Login';

describe('Login testing with Enzyme', () => {
  it('renders without crashing', () => {
    const wrapper = shallow(<Login />);

    expect(wrapper).toMatchSnapshot();
  });
});
