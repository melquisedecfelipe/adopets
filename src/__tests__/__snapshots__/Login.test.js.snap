// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login testing with Enzyme renders without crashing 1`] = `ShallowWrapper {}`;
